packages/framework/collabspaces/src/collabSpaces.ts

```diff
@@ -447,23 +457,42 @@ export class TempCollabSpaceRuntime
 
 	private cellChanged(row: number, col: number) {
 		const info = this.getCellInfo(row, col);
-		if (info.value !== undefined && isChannelDeffered(info.channelInfo?.type)) {
-			const channelId = info.channelId;
-			// Need to update channel and convert it to real thing.
-			const deferredChannel = this.deferredChannels.get(channelId);
-			assert(deferredChannel !== undefined, "deferred channel not found");
-			this.deferredChannels.delete(channelId);
 
-			this.destroyChannelCore(channelId);
+		if (info.value !== undefined) {
+			if (col === -1 || row === -1) {
+				// Col and Row == -1 means cell that uniquely identify the rows and columns from the matrix.
+				const mapInfo: { type: ReverseMapType; index: number } =
+					col === -1 ? { type: "row", index: row } : { type: "col", index: col };
+				this.reverseMap.addCellToMap(
```

Comment:
Let's add Pri1 comment in the code here to revisit how we raise matrix events.
Today we will raise "column added" event before we have an ID, and thus exposing incorrect state of matrix to consumers.
It's better if we do something like that (as a separate PR):
- suppress col/row addition events
- raise them here, when we learn about new ID.
It's not that simple, as proposed sequence would only work for a single row/col addition, not for many rows/columns added, but I'm sure we can figure out something.
Alternatively, we need better APIs from matrix...